keps/sig-api-machinery/2211-omit-managed-fields-server-side/README.md

```diff
@@ -0,0 +1,372 @@
+}
+
+func (s *serializer) Decode(data []byte, defaults *schema.GroupVersionKind, into runtime.Object) (runtime.Object, *schema.GroupVersionKind, error) {
+    return s.inner.Decode(data, defaults, into)
+}
+```
+
+3. Make `k8s.io/apimachinery/pkg/runtime/serializer.newSerializersForScheme` returns serializers for the new "unmanaged" content types.
+For instance:
+```go
+import (
+    ...
+    "k8s.io/apimachinery/pkg/runtime/serializer/unmanaged"
+)
+
+func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory, options CodecFactoryOptions) []serializerType {
+    ...
+    serializers := []serializerType{
+        jsonSerializerType,
+        {
+            AcceptContentTypes: []string{runtime.ContentTypeYAML},
+            ContentType:        runtime.ContentTypeYAML,
+            FileExtensions:     []string{"yaml"},
+            EncodesAsText:      true,
+            Serializer:         yamlSerializer,
+        },
+        {
+            AcceptContentTypes: []string{runtime.ContentTypeProtobuf},
+            ContentType:        runtime.ContentTypeProtobuf,
+            FileExtensions:     []string{"pb"},
+            Serializer:         protoSerializer,
+
+            Framer:           protobuf.LengthDelimitedFramer,
+            StreamSerializer: protoRawSerializer,
+        },
+    }
+    ...
+
+    allSerializers := append([]serializerType{}, serializers...)
+    for _, serializer := range serializers {
+        serializer.ContentType += runtime.ContentTypeSuffixOmitManagedFields
+        for i := range serializer.AcceptContentTypes {
+            serializer.AcceptContentTypes[i] += runtime.ContentTypeSuffixOmitManagedFields
+        }
+        serializer.Serializer = unmanaged.NewSerializer(serializer.Serializer)
+        serializer.StreamSerializer = unmanaged.NewSerializer(serializer.StreamSerializer)
+        serializer.PrettySerializer = unmanaged.NewSerializer(serializer.PrettySerializer)
+        allSerializers = append(allSerializers, serializer)
+    }
+    return allSerializers
+}
+```
+
+### Client-side
+
+<<[UNRESOLVED @knight42 ]>>
+I am uncertain about what the flag should look like in kubectl.
+<<[/UNRESOLVED]>>
+
+#### Alternative 1: New dedicated flag, for example `--show-managed-fields`
+
+The value of flag `--show-managed-fields` could be `omit`, `default` and a version name(in case users want to 
+request a specific version of managed fields), such as `v1`, `v2` etc, and the value defaults to `default`(do nothing).
+
+Pros:
+* It doesn't change the output format by default and leave users choices to omit the managedFields or request a specific version of them.
+
+Cons:
+* The new flag depends on the `-o` flag, it only works with `-o (json|yaml)`. In other cases, it is a no-op.
+
+#### Alternative 2: New output format, for example `-o yaml+managed.omit`
+
+Pros:
+* It leaves the current output formats intact and also enable users to choose to omit or request a certain version of managedFields.
```

Comment:
I guess you could permit both to work, but I don't really see the need for this if it's in kubeconfig and available via flag to override that.